backend/src/main/java/ca/mcgill/purposeful/controller/ReactionController.java

```diff
@@ -0,0 +1,55 @@
+package ca.mcgill.purposeful.controller;
+
+import ca.mcgill.purposeful.dto.ReactionDTO;
+import ca.mcgill.purposeful.exception.GlobalException;
+import ca.mcgill.purposeful.model.Reaction;
+import ca.mcgill.purposeful.model.Reaction.ReactionType;
+import ca.mcgill.purposeful.service.ReactionService;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * API for accessing the endpoints of Reaction
+ */
+@RestController
+@RequestMapping({"/api/reaction", "/api/reaction/"})
+public class ReactionController {
+
+  @Autowired
+  private ReactionService reactionService;
+
+  /**
+   * POST method to react
+   *
+   * @param reactionDTO - the reaction added or removed
+   * @return the newly created reaction DTO or the removed reaction DTO with null values
+   * @author Athmane Benarous
+   */
+  @PostMapping(value = {"/react", "/react/"})
+  @PreAuthorize("hasAnyAuthority('User', 'Moderator', 'Owner')")
```

Comment:
According to the domain model, reaction can only be associated to a regular user. Why are we authorizing moderator and owner?

Reply:
When a moderator or owner tries to react, it causes error 500. That is really bad since it means internal server error. The error should be 403 forbidden

Reply:
Btw, the reason it is 500 is because we are trying to create a reaction, but the regular user field is null (while being nullable false), since we are not passing a regular user

Reply:
I forgot to revert it to hasAuthority('User') after trying out many things to debug! It should indeed be restricted to users. Good catch

Reply:
Hasn't been fixed, I'll fix it right now